pos_reuse_pricelist/static/src/js/models.js

```diff
@@ -0,0 +1,27 @@
+/*
+ *  Copyright 2021 Akretion
+ *  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+ */
+
+odoo.define("pos_reuse_pricelist.models", function (require) {
+    "use strict";
+
+    var models = require("point_of_sale.models");
+    var posmodel_super = models.PosModel.prototype;
+
+    models.PosModel = models.PosModel.extend({
+        on_removed_order(removed_order, index, reason) {
+            posmodel_super.on_removed_order.apply(this, arguments);
+            var order_list = this.get_order_list();
+            if (
+                (reason === "abandon" || removed_order.temporary) &&
+                order_list.length > 0
+            ) {
+                // Do nothing when this is a deletion (and there is another order available)
```

Comment:
Better to inverse that `if` statement and remove this empty block

Reply:
The idea was to reuse the same `if` statement as the one in `posmodel_super.on_removed_order` to make it easier to understand and maintain.
I think the inversed statement may be harder to read

Reply:
Something like this maybe?
Could be accompanied with a comment

```js
    const newOrderCreated = !((reason === "abandon" || removed_order.temporary) && order_list.length > 0);
    if (newOrderCreated) {
        this.get_order().pricelist = removed_order.pricelist;
    }
```

Reply:
Good idea